tests/smoke-tests/test-dynamic-fees.ts

```diff
@@ -0,0 +1,241 @@
+      this.timeout(timeout);
+
+      runtime = context.polkadotApi.consts.system.version.specName.toUpperCase() as any;
+
+      const { specVersion, specName } = context.polkadotApi.consts.system.version;
+      if (specVersion.toNumber() < 2200) {
+        debug(`Runtime version ${specVersion.toString()} is less than 2200, skipping test suite.`);
+        this.skip();
+      }
+
+      if (specName.toString() !== "moonbase") {
+        debug(`Runtime ${specName.toString()} not supported by these tests, skipping.`);
+        this.skip();
+      }
+
+      const blockNumArray = await getBlockArray(context.polkadotApi, timePeriod, limiter);
+
+      debug(`Collecting ${hours} hours worth of block data`);
+
+      const getBlockData = async (blockNum: number) => {
+        const blockHash = await context.polkadotApi.rpc.chain.getBlockHash(blockNum);
+        const apiAt = await context.polkadotApi.at(blockHash);
+        const ethBlock = (await apiAt.query.ethereum.currentBlock()).unwrapOrDefault();
+        const ethersBlock = await context.ethers.getBlock(blockNum);
+        const weights = await apiAt.query.system.blockWeight();
+        return {
+          blockNum: blockNum,
+          nextFeeMultiplier: await apiAt.query.transactionPayment.nextFeeMultiplier(),
+          ethBlock,
+          ethersBlock,
+          baseFeePerGasInGwei: ethers.utils.formatUnits(ethersBlock.baseFeePerGas, "gwei"),
+          weights,
+        };
+      };
+
+      blockData = await Promise.all(
+        blockNumArray.map((num) => limiter.schedule(() => getBlockData(num)))
+      );
+    });
+
+    testIt("C100", "Block utilization by weight corresponds to fee multiplier", function () {
+      const maxWeights = context.polkadotApi.consts.system.blockWeights;
+      const enriched = blockData.map(({ weights, blockNum, nextFeeMultiplier }) => {
+        const fillPermill = weights.normal.refTime
+          .toBn()
+          .mul(new BN("1000000"))
+          .div(maxWeights.perClass.normal.maxTotal.unwrap().refTime.toBn());
+
+        const change = checkMultiplier(
+          blockData.find((a) => a.blockNum == blockNum - 1),
+          nextFeeMultiplier
+        );
+
+        return {
+          blockNum,
+          fillPermill,
+          change,
+          valid: isChangeDirectionValid(fillPermill, change, nextFeeMultiplier),
+        };
+      });
+
+      const failures = enriched.filter(({ valid }) => !valid);
+      failures.forEach(({ blockNum, fillPermill, change }) => {
+        debug(
+          `Block #${blockNum} is ${(fillPermill.toNumber() / 1_000_000).toFixed(
+            2
+          )}% full with feeMultiplier ${change}`
+        );
+      });
+
+      expect(
+        failures.length,
+        `Please investigate blocks ${failures.map(({ blockNum }) => blockNum).join(`, `)}`
+      ).to.equals(0);
+    });
+
+    testIt("C200", "Block utilization by gas corresponds to fee multiplier", async function () {
```

Comment:
It looks like this is expecting the block's gas utilization to correspond to changes in the multiplier, but this wouldn't work when there are any significant non-Ethereum Substrate extrinsics present.

Or am I missing something?